src/xcode/ENA/ENA/Source/Scenes/FileScanner/FileScannerCoordinatorViewModel.swift

```diff
@@ -27,46 +25,45 @@ class FileScannerCoordinatorViewModel: NSObject, PHPickerViewControllerDelegate,
 
 	@available(iOS 14, *)
 	func picker(_ picker: PHPickerViewController, didFinishPicking results: [PHPickerResult]) {
-		defer {
-			self.dismiss()
-		}
-
-		// each result represents a selected image
-		results.forEach { result in
-			let itemProvider = result.itemProvider
-			guard itemProvider.canLoadObject(ofClass: UIImage.self) else {
-				return
-			}
-			itemProvider.loadObject(ofClass: UIImage.self) { [weak self]  provider, _ in
-				guard let self = self,
-					  let image = provider as? UIImage,
-					  let codes = self.findQRCodes(in: image),
-					  !codes.isEmpty
-				else {
-					Log.debug("Looks like we have an issue reading the image", log: .fileScanner)
+		dismiss()
+		showHUD { [weak self] in
+			results.forEach { result in
+				let itemProvider = result.itemProvider
+				guard itemProvider.canLoadObject(ofClass: UIImage.self) else {
+					self?.qrCodesFound([])
 					return
 				}
-				self.qrCodesFound(codes)
+				itemProvider.loadObject(ofClass: UIImage.self) { [weak self]  provider, _ in
```

Comment:
This part looks similar to the part in `didFinishPickingMediaWithInfo`. Would be nice to have a new func and avoid code duplication.